sql/expression/function/extract_test.go

```diff
@@ -0,0 +1,224 @@
+			expected: 12,
+		},
+		{
+			name:     "get hour",
+			unit:     "HOUR",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 11,
+		},
+		{
+			name:     "get minute",
+			unit:     "MINUTE",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 22,
+		},
+		{
+			name:     "get second",
+			unit:     "SECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 33,
+		},
+		{
+			name:     "get microsecond",
+			unit:     "MICROSECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 445566,
+		},
+		{
+			name:     "get year_month",
+			unit:     "YEAR_MONTH",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 202311,
+		},
+		{
+			name:     "get day_hour",
+			unit:     "DAY_HOUR",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 1211,
+		},
+		{
+			name:     "get day_minute",
+			unit:     "DAY_MINUTE",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 121122,
+		},
+		{
+			name:     "get day_second",
+			unit:     "DAY_SECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 12112233,
+		},
+		{
+			name:     "get day_microsecond",
+			unit:     "DAY_MICROSECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 12112233445566,
+		},
+		{
+			name:     "get hour_minute",
+			unit:     "HOUR_MINUTE",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 1122,
+		},
+		{
+			name:     "get hour_second",
+			unit:     "HOUR_SECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 112233,
+		},
+		{
+			name:     "get hour_microsecond",
+			unit:     "HOUR_MICROSECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 112233445566,
+		},
+		{
+			name:     "get minute_second",
+			unit:     "MINUTE_SECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 2233,
+		},
+		{
+			name:     "get minute_microsecond",
+			unit:     "MINUTE_MICROSECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 2233445566,
+		},
+		{
+			name:     "get second_microsecond",
+			unit:     "SECOND_MICROSECOND",
+			dateTime: "2023-11-12 11:22:33.445566",
+			expected: 33445566,
+		},
+		{
+			name:     "get month 0",
+			unit:     "MONTH",
+			dateTime: "2023-00-12 11:22:33.445566",
+			expected: 0,
+			skip:     true,
+		},
+		{
+			name:     "get quarter 0",
+			unit:     "QUARTER",
+			dateTime: "2023-00-12 11:22:33.445566",
+			expected: 0,
+			skip:     true,
+		},
+		{
+			name:     "get day 0",
+			unit:     "DAY",
+			dateTime: "2023-01-00 11:22:33.445566",
+			expected: 0,
+			skip:     true,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.skip {
+				t.Skip()
+			}
+			require := require.New(t)
+			f := NewExtract(expression.NewLiteral(tt.unit, types.LongText), expression.NewLiteral(tt.dateTime, types.LongText))
+			v, err := f.Eval(sql.NewEmptyContext(), nil)
+			require.NoError(err)
+			require.Equal(tt.expected, v)
+		})
+	}
+
+	t.Run("test extract null datetime", func(t *testing.T) {
```

Comment:
Would be good to have a test for a bad date, too. Looks like MySQL returns `NULL` and emits a warning about truncation. At minimum, would be good to test for `NULL` return value for a bad date, but checking for the warning, too, would be nice, although I suspect that needs to happen in the Date Convert function.